qiita_pet/static/js/sampleTemplateVue.js

```diff
@@ -435,6 +463,26 @@ Vue.component('sample-template-page', {
         // Add the select to update the sample information
         $row = $('<div>').attr('id', 'update-st-div').addClass('row form-group').appendTo($tab);
         $('<label>').addClass('col-sm-2 col-form-label').append('Update sample information:').appendTo($row);
+
+        // Add the direct upload field
+        $('<label>')
+          .addClass('btn btn-default')
+            .append('Direct upload file <small>(< 2M)</small>')
+            .appendTo($row)
+            .append('<input type="file" style="display: none;" id="st-direct-upload">');
+        $('#st-direct-upload').on('change', function() {
+          if (this.files.length != 1) {
+            alert('You can only upload one file.')
+            return false;
+          }
+          if (this.files[0].size > 2097152) {
+            alert('You can only upload files smaller than 2M; please use the regular uploader.');
```

Comment:
2M -> 2MB. A newcomer might wonder what is the regular uploader, what about linking to it?

Reply:
good point, note that afaik a basic alert doesn't have clickable section; anyway, re-writing to make clearer.